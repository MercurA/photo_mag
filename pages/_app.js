import '../styles/globals.css'
import { createContext, useReducer } from "react"
import reducer, { initialState } from "../utils/state/reducer"

export const AppContext = createContext(null)

const MyApp = ({ Component, pageProps }) => {
  const [state, dispatch] = useReducer(reducer, initialState)

  return (
    <AppContext.Provider value={{ ...state, dispatch }}>
      <Component {...pageProps} />
    </AppContext.Provider>
  )
}

export default MyApp