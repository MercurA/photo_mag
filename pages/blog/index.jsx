import DesktopBlogContainer from '../../containers/DesktopBlogContainer'

const Blog = () => {
    return (
        <DesktopBlogContainer />
    )
}

export default Blog;