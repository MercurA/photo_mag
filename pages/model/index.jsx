import { useRef, useState } from 'react'
import styles from './model.module.css'
import { Canvas, useFrame } from '@react-three/fiber'

const Sphere = ({position, size, color}) => {
    const [hovered, setHovered] = useState(false)
    const sphereRef = useRef()
    useFrame((state, delta) => {
        sphereRef.current.rotation.y += delta/6
    })
    return (
        <mesh ref={sphereRef} onPointerOver={(e) => {e.stopPropagation(); setHovered(!hovered)}}>
            <sphereGeometry />
            <meshStandardMaterial color={hovered ? color : 'lightblue'} wireframe/>
        </mesh>
    )
}

const Model = () => {
    return (
        <div className={styles.container}>
            <Canvas >
                <directionalLight position={[0, 0, 7]} />
                <Sphere position={[0,0,0]} size={[1,1,1]} color={'orange'}/>
            </Canvas>
        </div>
    )
}

export default Model