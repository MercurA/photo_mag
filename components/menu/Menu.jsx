import { useContext } from 'react'

import styles from './menu.module.css'
import Drawer from './Drawer'
import { ACTIONS } from '../../utils/state/reducer'
import { AppContext } from '../../pages'

const Menu = ({dispatch}) => {
    const context = useContext(AppContext)

    const {isDrawerOpen, isPopupOn} = context

    const handleDrawer = () => {
        dispatch({type: ACTIONS.drawer})
    }

    return (
        <div className={`${styles.menuContainer} ${isPopupOn && styles.hideMenu}`}>
            <div className={styles.menuWrapper} onClick={handleDrawer}>
                <div className={styles.slideMenuBtn}>
                    <span></span>
                    <span></span>
                    <span></span>
                </div>
                <div className={styles.brandTitle}>
                    <strong>{"Horea's"}</strong> Photography
                </div>
            </div>
            {/* Drawer */}
            <Drawer isOpen={isDrawerOpen} handleDrawer={handleDrawer}/>
        </div>
    )
}

export default Menu