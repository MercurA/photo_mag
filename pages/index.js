import { createContext, useReducer } from "react";
import CardContainer from "../components/card_image/CardContainer";
import Menu from "../components/menu/Menu";
import MusicContainer from "../components/music/MusicContainer";
import reducer, { initialState } from "../utils/state/reducer";

export const AppContext = createContext(null)

export default function Home() {
  const [state, dispatch] = useReducer(reducer, initialState)

  return (
    <AppContext.Provider value={state}>
      <div >
        <Menu dispatch={dispatch}/>
        <CardContainer />
        <MusicContainer dispatch={dispatch}/>
      </div>
    </AppContext.Provider>
  )
}
