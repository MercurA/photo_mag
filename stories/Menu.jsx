import './menu.css'
import React from 'react';

export const Menu = ({}) => {

    const handleDrawer = () => {}

    return (
        <div className={'menuContainer'}>
            <div className={'menuWrapper'} onClick={handleDrawer}>
                <div className={'slideMenuBtn'}>
                    <span></span>
                    <span></span>
                    <span></span>
                </div>
                <div className={'brandTitle'}>
                    <strong>{"Horea's"}</strong> Photography
                </div>
            </div>
        </div>
    )
}
