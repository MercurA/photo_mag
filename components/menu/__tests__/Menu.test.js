import { render, screen } from '@testing-library/react'
import Menu from '../Menu'
import { AppContext } from '../../../pages'

describe('Menu', () => {
    let state = {
        isDrawerOpen: false,
        isPopupOn: false
    }
    it('should render corectly', () => {
        const comp = render(
            <AppContext.Provider value={state}>
                <Menu />
            </ AppContext.Provider>
        )

        expect(comp).toMatchSnapshot()
    })
    it('should render corectly with drawer open', () => {
        state = {
            isDrawerOpen: true,
            isPopupOn: false
        }
        const comp = render(
            <AppContext.Provider value={state}>
                <Menu />
            </ AppContext.Provider>
        )

        expect(comp).toMatchSnapshot()
    })
})