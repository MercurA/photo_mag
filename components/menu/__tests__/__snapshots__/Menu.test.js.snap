// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render corectly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="menuContainer false"
      >
        <div
          class="menuWrapper"
        >
          <div
            class="slideMenuBtn"
          >
            <span />
            <span />
            <span />
          </div>
          <div
            class="brandTitle"
          >
            <strong>
              Horea's
            </strong>
             Photography
          </div>
          <div
            class="container false"
          >
            <div
              class="stopped"
            >
              <img
                alt="stop music icon"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="/icons/music-stop.svg"
                style="color: transparent;"
                width="20"
              />
            </div>
            <audio
              autoplay=""
              id="audio"
              loop=""
              src="/media/trance1.flac"
            />
          </div>
        </div>
        <div
          class="drawer drawerClosed"
        >
          <div
            class="drawerHeader"
          >
            <button
              class="drawerCloseBtn"
            >
              <img
                alt="close icon"
                data-nimg="1"
                decoding="async"
                height="15"
                loading="lazy"
                src="/icons/close.svg"
                style="color: transparent;"
                width="15"
              />
            </button>
          </div>
          <div
            class="drawerBody"
          >
            <div
              class="drawerItem"
            >
              Galery
            </div>
            <div
              class="drawerItem"
            >
              Shop
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="menuContainer false"
    >
      <div
        class="menuWrapper"
      >
        <div
          class="slideMenuBtn"
        >
          <span />
          <span />
          <span />
        </div>
        <div
          class="brandTitle"
        >
          <strong>
            Horea's
          </strong>
           Photography
        </div>
        <div
          class="container false"
        >
          <div
            class="stopped"
          >
            <img
              alt="stop music icon"
              data-nimg="1"
              decoding="async"
              height="20"
              loading="lazy"
              src="/icons/music-stop.svg"
              style="color: transparent;"
              width="20"
            />
          </div>
          <audio
            autoplay=""
            id="audio"
            loop=""
            src="/media/trance1.flac"
          />
        </div>
      </div>
      <div
        class="drawer drawerClosed"
      >
        <div
          class="drawerHeader"
        >
          <button
            class="drawerCloseBtn"
          >
            <img
              alt="close icon"
              data-nimg="1"
              decoding="async"
              height="15"
              loading="lazy"
              src="/icons/close.svg"
              style="color: transparent;"
              width="15"
            />
          </button>
        </div>
        <div
          class="drawerBody"
        >
          <div
            class="drawerItem"
          >
            Galery
          </div>
          <div
            class="drawerItem"
          >
            Shop
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Menu should render corectly with drawer open 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="menuContainer false"
      >
        <div
          class="menuWrapper"
        >
          <div
            class="slideMenuBtn"
          >
            <span />
            <span />
            <span />
          </div>
          <div
            class="brandTitle"
          >
            <strong>
              Horea's
            </strong>
             Photography
          </div>
          <div
            class="container hideIcon"
          >
            <div
              class="stopped"
            >
              <img
                alt="stop music icon"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="/icons/music-stop.svg"
                style="color: transparent;"
                width="20"
              />
            </div>
            <audio
              autoplay=""
              id="audio"
              loop=""
              src="/media/trance1.flac"
            />
          </div>
        </div>
        <div
          class="drawer drawerOpen"
        >
          <div
            class="drawerHeader"
          >
            <button
              class="drawerCloseBtn"
            >
              <img
                alt="close icon"
                data-nimg="1"
                decoding="async"
                height="15"
                loading="lazy"
                src="/icons/close.svg"
                style="color: transparent;"
                width="15"
              />
            </button>
          </div>
          <div
            class="drawerBody"
          >
            <div
              class="drawerItem"
            >
              Galery
            </div>
            <div
              class="drawerItem"
            >
              Shop
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="menuContainer false"
    >
      <div
        class="menuWrapper"
      >
        <div
          class="slideMenuBtn"
        >
          <span />
          <span />
          <span />
        </div>
        <div
          class="brandTitle"
        >
          <strong>
            Horea's
          </strong>
           Photography
        </div>
        <div
          class="container hideIcon"
        >
          <div
            class="stopped"
          >
            <img
              alt="stop music icon"
              data-nimg="1"
              decoding="async"
              height="20"
              loading="lazy"
              src="/icons/music-stop.svg"
              style="color: transparent;"
              width="20"
            />
          </div>
          <audio
            autoplay=""
            id="audio"
            loop=""
            src="/media/trance1.flac"
          />
        </div>
      </div>
      <div
        class="drawer drawerOpen"
      >
        <div
          class="drawerHeader"
        >
          <button
            class="drawerCloseBtn"
          >
            <img
              alt="close icon"
              data-nimg="1"
              decoding="async"
              height="15"
              loading="lazy"
              src="/icons/close.svg"
              style="color: transparent;"
              width="15"
            />
          </button>
        </div>
        <div
          class="drawerBody"
        >
          <div
            class="drawerItem"
          >
            Galery
          </div>
          <div
            class="drawerItem"
          >
            Shop
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
